docs/nGQL-tutorial.md

```diff
@@ -0,0 +1,259 @@
+
+|    | CREATE | DROP | ALTER | DESCRIBE | SHOW | TTL | LOAD | DUMP |
+|:-: | :-: | :-: |:-: | :-: | :-: | :-: |:-: | :-: |
+|TAG | √      | v0.2 |    v0.2  |      √   |   √  |  v0.3  | v0.2    |  v0.3   |
+|EDGE| √      |v0.2  |  v0.2 |  √       |  √   |v0.3 | v0.2 | v0.3 |
+
+You can use CREATE, DROP, ALTER, DESCRIBE to create, drop, alter, view a schema.
+Following are some examples:
+
+```
+CREATE TAG player(name string, age int);
+```
+
+```
+DESCRIBE TAG player;
+```
+
+```
+CREATE TAG team(name string);
+```
+
+```
+DESCRIBE TAG team;
+```
+
+```
+CREATE EDGE serve (start_year int, end_year int);
+```
+
+```
+DESCRIBE EDGE serve;
+```
+
+```
+CREATE EDGE like (likeness double)；
+```
+
+```
+SHOW TAGS;
+```
+
+```
+SHOW EDGES
+```
+
+### Data manipulation
+
+INSERT is used to insert new vertices and edges, UPDATE AND REMOVE will be available in v0.2.
+
+|   | INSERT | UPDATE | REMOVE |
+|:-: | :-: | :-: |:-: |
+|TAG | √   | v0.2     | v0.2   |
+|EDGE | √   | v0.2    | v0.2|
+
+When inserting a vertex, its tag type and ID should be specified, but the attribute field can be ignored(the ignored fields are set to default values).
+
+Following are some examples:
+
+```
+INSERT VERTEX player(name, age) VALUES 100:("Stoudemire", 36);
+```
+
+```
+INSERT VERTEX player(name) VALUES 101:("Vicenta"); -- age is set to default value 0
+```
+
+```
+INSERT VERTEX player(name) VALUES 102:("jummy");
+```
+
+```
+INSERT VERTEX team(name) VALUES 201:("Magic");
+```
+
+```
+INSERT EDGE like (likeness) VALUES 100 -> 101:(90.02);
+```
+
+```
+INSERT EDGE like (likeness) VALUES 101 -> 102:(10);
+```
+
+```
+INSERT EDGE serve (start_year, end_year) VALUES 101 -> 201:(2002, 2010);
+```
+
+### Graph query
+
+The most commonly used graph query/traversal operator is GO, it means starting from a certain point and querying its 1 degree neighbor. Complex queries can be done by combining pipe `|`, filtering `WHERE`, `YIELD`, etc.
+
+Following are some examples:
+
+```
+GO FROM 100 OVER like; -- Start from vertex 100, query 1-hop along edge like.
+```
+
+```
+GO 2 STEPS FROM 100 OVER like; -- Start from vertex 100, query 2-hop along edge
+```
+
+```
+GO FROM 100 OVER like WHERE likeness >= 0; -- Start from vertex 100, query along edge like and filter its property likeness
+```
+
+```
+GO FROM 100 OVER like WHERE $$[player].name=="Vicenta"; -- Filter requirement: the destination vertex name is "Vicenta"
+```
+
+```
+GO FROM 101 OVER serve YIELD serve._src AS src_id, $^[player].age AS src_propAge, serve._dst AS dst_id, $$[team].name AS dst_propName; -- Return the starting vertex id(renamed as srcid), source vertex property age, destination vertex id and its name
+```
+
+```
+GO FROM 100 OVER like | GO FROM $-.id OVER serve; -- Start from vertex 100, query 1-hop, set its output as the next query's input by using pipe
+```
+
+
+
+## Syntax norms
```

Comment:
@dutor 
what do u think about nGQL code style recommendation?
My thoughts:
1. KEYWORDS are UPPER. e.g.,  GO FROM  
2. all others are camel. e.g., edgeName, colName, tagName, propName.
2' or edge_name, col_name, tag_name, prop_name.   //either is OK
3.  "\" for a newline
4. using -- for comment.